Changes

kduncanwelke opened the pull request:
Fixes #15925

To test: In old code, set device text side under accessibility to use large text (use a high setting). Open the app and try to add a new site - with large text, the search field for a url is covered by the keyboard.

In this edited code, the table has had an offset applied, which pushes the search field up above the keyboard, making it visible. This code will apply if the table exceeds the height of the device, like on small devices or if the text in the site creation header is very large. It will be ignored on iPads/bigger devices.

## Regression Notes
1. Potential unintended areas of impact
The code could have applied erroneously when not needed, or caused an increasing offset.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested on different devices sizes to make sure this did not happen, and tested going back and forth between views and hiding and showing the keyboard.

3. What automated tests I added (or what prevented me from doing so)
None, as the success of the fix depended very much on the visual, as confirmed by manual testing and eyeballing.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

kduncanwelke committed:
Fix to url search field bug

kduncanwelke committed:
Removed old code, added comments

chipsnyder commented:
Hi @kduncanwelke, 

I just came across this PR for https://github.com/wordpress-mobile/WordPress-iOS/issues/15925. Thank you so much for proposing a fix. I'll run this fix today and leave some code review notes. 

I see that our automated bot called out a few style issues. You should be able to correct all of these with `rake lint:autocorrect`

chipsnyder commented:
Hey @kduncanwelke 👋 

Thanks again for taking a look at this issue. I was wondering if you could help me connect the pieces. I wasn't able to locate where this class was being used on the screen we're targeting. It seems as if it might not be referenced but perhaps there is something I'm missing?

kduncanwelke commented:
Heyo @chipsnyder thank you for looking into this! It has been some months since I touched this issue, but from what I recall it took some time to figure out that this class was responsible. I am cloning the project (new machine) and will fix the style issues and get back to you as soon as possible on how this fix connects with the issue.

kduncanwelke committed:
Fixed whitespace

kduncanwelke commented:
Hi again @chipsnyder - I have fixed the whitespacing issue in the repo, so hound should like it now. 

I am unfortunately unable to get to the bottom of where TableViewOffsetCoordinator is implemented, as I am on an M1 computer, which is not compatible with the project. What I suspect, however, is that it is implemented in one of the included WordPress frameworks, as this was the case with a previous accessibility site creation issue I fixed (the problem was fixed in WordPress Authenticator).

Changes to TableViewOffsetCoordinator do have the desired effect in fixing the problem, so it is responsible, but I'm afraid with the M1 not having support for this project yet, I cannot point out directly where the implementation is. 😓

chipsnyder commented:
Hi @kduncanwelke Thanks for cleaning up those formatting issues. 

> as I am on an M1 computer, which is not compatible with the project.

We did finally add support for M1 devices so you may just need to update your fork (I'm on a new M1 iMac and I was able to get your branch up and running after merging in the latest `develop`).

I'd be happy to help with any errors you might be having. 

----- 

Unfortunately, these changes don't appear to be working for me 😞 this is what I see when I run your branch updated with `develop`

https://user-images.githubusercontent.com/3384451/126842204-305b02d1-8e5c-4e0a-b430-c5e12908a6ac.mp4

We have been doing a lot of work in this area though so potentially your change was more closely tied to an area that was refactored.

I was focused on some bug fixes this week so I took a stab at this one as well. If you'd like to compare notes here is the path I took https://github.com/wordpress-mobile/WordPress-iOS/pull/16923

kduncanwelke commented:
@chipsnyder Ah, thank you for the info about the M1 now working with the project! Mine must be rather out of date. 😅

I am going to ahead and close this pr, as the changes aren't performing as expected any more, per the video, and it sounds like you have it in hand already. Thank you for taking the time to check this pr out - it's just not a good fit with where the code has evolved to now. 👍🏻
